ui/hooks/gasFeeInput/useTransactionMetrics.js

```diff
@@ -0,0 +1,29 @@
+import { useContext } from 'react';
+
+import { MetaMetricsContext } from '../../contexts/metametrics';
+
+export const useTransactionMetrics = ({ transaction }) => {
+  const metricsEvent = useContext(MetaMetricsContext);
+  const { origin, type } = transaction;
+
+  const captureTransactionMetrics = ({ action, name, variables }) => {
+    metricsEvent({
+      eventOpts: {
+        category: 'Transactions',
+        action,
+        name,
+      },
+      customVariables: {
+        origin,
+        transaction_type: type,
+        source: origin === 'metamask' ? 'user' : 'dapp',
+        EIP_1559_V2: true,
```

Comment:
This function seems like it could be pretty useful in general, but tying it specifically to EIP-1559 V2 could be a surprise for people. What if this was an explicit thing? So within EIP-1559 V2 features, we say `captureTransactionMetrics({ action: ..., name: ..., EIP_1559_V2: true })`, but in other places we might just say `captureTransactionMetrics({ action: ..., name: ... })`.

Reply:
This is mostly used in V2 implementation, `variables` can be used to override this.

Reply:
I see. It may be getting used in the EIP-1559 V2 code, but I'm not sure everyone knows that. The name of this function doesn't say anything about where it should be used or what it does by default, so it seems like it could be misleading. One might reasonably assume in fact that EIP-1559 V2 is _not_ the default considering that it's not the default elsewhere (yet).

If you want to keep this line, then what do you think about calling this function `useTransactionMetricsForEIP1559V2` or something and have `EIP_1559_V2: true` be non-overridable? Then if we want to use this in a more general way we can come back later and make a decision about this.

Reply:
I changed it to 2 methods `captureTransactionMetrics` which is more generic and `captureTransactionMetricsForEIP1559V2` which uses the previous one.

Reply:
Thanks!